### PR TITLE
Updating base docker image to alpine.

### PR DESCRIPTION
**Pull request by vishh**
cc @mikedanese: I assume this will actually resolve the cert issues.

### CODE REVIEW

**Comment by mikedanese** [comment]
but does it break dns-search paths? @thockin what should be aware of when switching to alpine?

**Comment by jimmidyson** [comment]
Yes it will. Install glibc & you still get a small image but with expected DNS behaviour. See https://github.com/anapsix/docker-alpine-java/blob/master/Dockerfile#L8 for how to do it.

**Comment by vishh** [comment]
Thanks for the tip @jimmidyson! Updated the PR. PTAL.

File: deploy/docker/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
-FROM progrium/busybox
+FROM alpine:3.2
 MAINTAINER vishnuk@google.com
 
-# cAdvisor discovery via external files.
-VOLUME /var/run/heapster/hosts
-
-RUN opkg-install ca-certificates
+RUN apk --update add curl ca-certificates tar && \
+    curl -Ls https://circle-artifacts.com/gh/andyshinn/alpine-pkg-glibc/6/artifacts/0/home/ubuntu/alpine-pkg-glibc/packages/x86_64/glibc-2.21-r2.apk > /tmp/glibc-2.21-r2.apk && \
```
Comment (jimmidyson) [inline deploy/docker/Dockerfile]: I guess you haven't tried building this? That URL doesn't work IIRC. Try changing the 6 to 8. Relying on build artifacts isn't great but no other option right now if using alpine.

Comment (vishh) [inline deploy/docker/Dockerfile]: I did try building and running the image. The e2e test also builds it and runs it in a cluster.

Comment (jimmidyson) [inline deploy/docker/Dockerfile]: Interesting - failed for me, sorry. Most have been a local issue.

**Comment by mikedanese** [comment]
LGTM.

File: deploy/docker/Dockerfile (hunk 1 above)
Comment (vishh) [inline deploy/docker/Dockerfile]: No problem :) Updated to `8` as suggested.

**Comment by vishh** [comment]
Merging this. @jimmidyson If there are any other issues, I will address them in a separate PR.

**Comment by thockin** [comment]
I will take a look at this - how big is an Alpine+glibc base?  I am
concerned that people will see this and use alpines package manager which
is just not going to work.

File: Makefile [hunk 2]
```diff
@@ -8,7 +8,7 @@ deps:
 	go get github.com/tools/godep
 
 build: clean deps
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build -a ./...
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build -a ./...
```
Comment (thockin) [inline Makefile]: trailing whitespace....

**Pull request merged**